src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/CompletionIntegrationTests.cs

```diff
@@ -457,4 +468,41 @@ private async Task CommitCompletionAndVerifyAsync(string expected, char? commitC
         // tests allow for it as long as the content is correct
         AssertEx.AssertEqualToleratingWhitespaceDifferences(expected, text);
     }
+
+    private async Task CommitCompletionAndVerifyAsync(string expected, string expectedSelectedItemLabel, char? commitChar = null)
+    {
+        // Actually open completion UI and wait for it have selected item we are interested in
+        var session = await TestServices.Editor.OpenCompletionSessionAndWaitForItemAsync(TimeSpan.FromSeconds(10), expectedSelectedItemLabel, HangMitigatingCancellationToken);
+
+        Assert.NotNull(session);
+        if (commitChar.HasValue)
+        {
+            // Commit using the specified commit character
+            session.Commit(commitChar.Value, HangMitigatingCancellationToken);
+
+            // session.Commit call above commits as if the commit character was typed,
+            // but doesn't actually insert the character into the buffer.
+            // So we still need to insert the character into the buffer ourselves.
+            TestServices.Input.Send(commitChar.Value.ToString());
+        }
+        else
+        {
+            Assert.True(session.CommitIfUnique(HangMitigatingCancellationToken));
+        }
+
+        var textView = await TestServices.Editor.GetActiveTextViewAsync(HangMitigatingCancellationToken);
+
+        var stopwatch = new Stopwatch();
+        string text;
+        while ((text = textView.TextBuffer.CurrentSnapshot.GetText()) != expected && stopwatch.ElapsedMilliseconds < 10000)
```

Comment:
Is it worth adding a constant for `10000` or comparing `stopwatch.Elapsed` against `TImeSpan.FromSeconds(10)`?

Reply:
Added a constant.